app/src/test/java/com/weatherxm/util/NumberUtilsTest.kt

```diff
@@ -43,19 +58,74 @@ class NumberUtilsTest : BehaviorSpec({
                 then("compactNumber should return a valid number") {
                     NumberUtils.compactNumber(100) shouldBe "100"
                 }
-                then("formatNumber should return a valid number") {
-                    NumberUtils.formatNumber(100) shouldBe "100"
+                and("it's a number containing decimals") {
+                    then("formatNumber should return a valid number") {
+                        NumberUtils.formatNumber(100.4, 1) shouldBe "100.4"
+                    }
+                }
+                and("it's a number containing thousands") {
+                    then("formatNumber should return a valid number") {
+                        NumberUtils.formatNumber(1000.4, 1) shouldBe "1,000.4"
+                    }
                 }
                 then("roundToDecimals should return a valid number") {
                     NumberUtils.roundToDecimals(100.123456) shouldBe 100.1F
                     NumberUtils.roundToDecimals(100.15) shouldBe 100.2F
                     NumberUtils.roundToDecimals(100.02) shouldBe 100.0F
                 }
-                then("roundToInt should return a valid number") {
-                    NumberUtils.roundToInt(1.0) shouldBe 1
-                    NumberUtils.roundToInt(1.49) shouldBe 1
-                    NumberUtils.roundToInt(1.5) shouldBe 2
-                    NumberUtils.roundToInt(1.51) shouldBe 2
+            }
+        }
+    }
+
+    context("Formatting tokens as a text with min 2 and max 3 decimals") {
+        given("a token amount") {
+            When("it is null") {
+                then("the formatter should return an empty string") {
+                    formatTokens(null) shouldBe EMPTY_VALUE
+                }
+            }
+            When("it is zero") {
+                then("the formatter should return 0.00") {
+                    formatTokens(0F) shouldBe "0.00"
+                }
+            }
+            When("it is an integer") {
+                and("it is == 1000") {
+                    then("return the integer with thousands separator and 2 decimals") {
+                        formatTokens(1000F) shouldBe "1,000.00"
+                    }
+                }
+                and("it is < 1000") {
+                    then("the formatter should return the integer with 2 decimals e.g. 10.00") {
+                        formatTokens(10F) shouldBe "10.00"
+                    }
+                }
+            }
+            When("it has up to 2 decimals") {
+                then("the formatter should return the amount with 2 decimals") {
+                    formatTokens(10.1F) shouldBe "10.10"
+                    formatTokens(10.01F) shouldBe "10.01"
+                }
+            }
+            When("it has >=3 decimals") {
+                then("the formatter should return the amount with 2 decimals") {
+                    formatTokens(10.001F) shouldBe "10.00"
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/WeatherXM/wxm-android/pull/351#issuecomment-2388460755>#323277</a><span> - 10/02/2024, 12:21 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The test case using 10.001F as input may produce inconsistent results due to floating-point precision issues.
</div>
<div id="fix">
<b>Fix:</b> Replace the use of Float with BigDecimal for more precise decimal calculations, especially when dealing with financial data.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -111,7 +111,7 @@ class NumberUtilsTest : BehaviorSpec({
          When("it has >=3 decimals") {
              then("the formatter should return the amount with 2 decimals") {
 -                formatTokens(10.001F) shouldBe "10.00"
 +                formatTokens(BigDecimal("10.001")) shouldBe "10.00"
                  formatTokens(10.005F) shouldBe "10.01"
              }
          }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged